pyiceberg/io/pyarrow.py

```diff
@@ -1293,8 +1371,8 @@ def field_partner(self, partner_struct: Optional[pa.Array], field_id: int, _: st
 
             if isinstance(partner_struct, pa.StructArray):
                 return partner_struct.field(name)
-            elif isinstance(partner_struct, pa.Table):
-                return partner_struct.column(name).combine_chunks()
+            elif isinstance(partner_struct, pa.RecordBatch):
+                return partner_struct.column(name)
```

Comment:
I know that it wasn't there before, but should we throw if we find something other than a `RecordBatch` or `StructArray`?